### DISCUSSION TITLE
Reducing android ANRs by switching from native Sound to an external ANE

### DISCUSSION THREAD

**Discussion by shaymolho** [airsdk/Adobe-Runtime-Support#1719] [Q&A] [answered]
Hello, as per the last comment in the discussion https://forum.starling-framework.org/d/23465-help-with-high-anr-rate/28
There's a recommendation to  move to MediaPlayer ANE

> Music playback with flash. Use external ANEs to play music. Sounds were less of an issue as they end quickly. MediaPlayer ANE works great as ANR fix.

Is this recommendation still relevant now? In my intuition adding an external ANE should increase ANR and not decrease it
@ajwfrost What's your take?

**Comment by ajwfrost** [comment, marked as answer]
The runtime on Android has gone through a few iterations of how the sound system works ... it looked like before AIR 32 there were some major changes and we've continued with this new approach, but have still found either crashes or hangs happening because of the complex multithreading and JNI combination that's involved in this...

So I can see that using an ANE which has a simpler mechanism/interface would likely make it more stable. The fact that the code is very multi-purpose and multi-platform at its core means that there are some complexities in AIR that wouldn't exist in a specialist media player engine for Android.

But, we are hoping to update this (or rather, to introduce some new APIs that allow you to deal at a higher level with common sound-based requirements such as "play this file/blob") which should help reduce the ANRs and instabilities that are happening in the existing code.. it's a bit delayed though due to other challenges/priorities so it may be worth going the ANE route to start with.

thanks
